components/experimental/src/measure/measureunit.rs

```diff
@@ -88,13 +89,11 @@ impl MeasureUnit {
 
             if zeros_count > 3 {
                 let significant_digits = &result.split_at(result.len() - zeros_count).0;
-                append_to.push_str(significant_digits);
-                append_to.push('E');
-                append_to.push_str(&zeros_count.to_string());
+                write!(buff, "{}E{}", significant_digits, zeros_count).unwrap();
```

Comment:
nit: write variables in the format string so that it's clearer what goes where
```suggestion
                write!(buff, "{significant_digits}E{zeros_count}").unwrap();
```